### PR TITLE
Milestone 2: Components and Redux store

### PR DESCRIPTION
**Pull request by abruzy**
## React Bookstore Milestone 2: Components and Redux store

In this milestone I:

- prepare the folder structure
- implemented App component to render BookList and BookForm component
- implemented BookList component to display books as an HTML table and each row have Book ID, title and category
- implemented Book component to accept book as props and represent a single row in Books table
- implemented a BookForm component to render a simple form for adding books, which contain a book title input, a select box with categories, and a submit button.
- implemented redux action, reducers, and store

### CODE REVIEW

File: src/reducers/books.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+const BooksArray = [
+  { id: randomId(), title: 'Do androids dream of electric sheep?', category: 'Action' },
+  { id: randomId(), title: 'Eveything I eve told you', category: 'Biography' },
+  { id: randomId(), title: 'Are you there', category: 'History' },
+  { id: randomId(), title: 'The devils wears', category: 'Horror' },
+  { id: randomId(), title: 'The curious incident', category: 'Kids' },
+  { id: randomId(), title: 'How to win friends', category: 'Learning' },
+  { id: randomId(), title: 'Cloudy with a chance', category: 'Sci-Fi' },
+];
```
Comment (bellom) [inline src/reducers/books.js:11]: I think this should be inside `index.js`

File: src/index.js [hunk 2]
```diff
@@ -1,11 +1,18 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
 import './index.css';
-import App from './App';
+import { createStore } from 'redux';
+import { Provider } from 'react-redux';
+import App from './components/App';
+import rootReducer from './reducers';
+
```
Comment (bellom) [inline src/index.js:8]: Let's bring up our `BooksArray` in here.
image
https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896827/

**Comment by abruzy** [comment]
> ## Hi @abruzy 
> ### Awesome job so far!! 
> **Quick One:** _**@abruzy this project is one of the collaborative projects but you're working alone, may I ask why. kindly drop your answer in here for other TSE to see, thanks.**_
> 
> #### You have done wonderfully well, you passed `almost` all the checklist for this project  just a few issues left to work on............ 
> **Kindly check out my comments below;**
> 
> ###  Kindly re-submit for review after making all those changes  Thumps -up await you.....
> **Best,**
> 
> **Oluwaseun Bello**
> **_Slack Handle - @sean bellom_**

I didn't get a partner for my React section, so I was told to work alone

**Comment by abruzy** [comment]
> ## Hi @abruzy 
> ### Awesome job so far!! 
> **Quick One:** _**@abruzy this project is one of the collaborative projects but you're working alone, may I ask why. kindly drop your answer in here for other TSE to see, thanks.**_
> 
> #### You have done wonderfully well, you passed `almost` all the checklist for this project  just a few issues left to work on............ 
> **Kindly check out my comments below;**
> 
> ###  Kindly re-submit for review after making all those changes  Thumps -up await you.....
> **Best,**
> 
> **Oluwaseun Bello**
> **_Slack Handle - @sean bellom_**

I didn't get a partner for my React section, so I was told to work alone

**Pull request merged**